internal/daemon/device-handler/dpu-device-handler/dpudevicehandler.go

```diff
@@ -86,12 +86,6 @@ func (d *dpuDeviceHandler) SetupDevices() error {
 
 	defer close(d.setupDevicesDone)
 
-	// Currently NF devices do not require any setup outside the VSP
-	if d.dpuMode {
-		d.log.Info(("Dpu mode detected, skipping devHandler devices setup"))
-		return nil
-	}
```

Comment:
It doesn't seem right to proceed to setup VFs (at least not for some kinds of DPU).

Reply:
```
	numVfs, err := d.vsp.SetNumVfs(8)
	if err != nil {
		// Currently NF devices do not require any setup outside the VSP
		// ignore the error if we are in DPU mode.
		if d.dpuMode {
			d.log.Info("Failed to set sriov numVFs, but ignoring error in DPU mode", "error", err)
		} else {
			return fmt.Errorf("failed to set sriov numVfs: %v", err)
		}
	} else {
		if numVfs.VfCnt == 0 {
			return fmt.Errorf("SetNumVfs ran, but numVfs == %d", numVfs.VfCnt)
		}
		d.log.Info("Num VFs set by VSP", "vf_count", numVfs.VfCnt)
	}
```
Below explains this. The API should be generic for all DPUs. And the API must be called, it is up to the VSP to properly respond back.